### ISSUE TITLE
Add Consul based service discovery

### ISSUE THREAD

**Issue by mlosiewicz-pl** [akka/akka-management#198]
I would like to add Consul based Service Discovery plugin and since I have a working implementation (although it's very specific and designed for our case) I would like to discuss the case.

The most basic approach would be to base the search on service name (defaulting to `ActorSystem` name in Service Discovery) but if you are using cluster roles then probably you register every node under different service name in consul (i.e. public API might be registered under a name that might be used by Consul Template to generate LB configuration and nodes responsible for processing would be registered under different names). Considering that I would suggest using tags in registered services and base Service Discovery on that. 

Another issue is the fact that Service Discovery requires access to node's Akka Management endpoint and it might not be the exposed under the same port as the registered HTTP check for the service in Consul, so also this should be IMO done on tags.

To sum up, I would like to propose a plugin for Consul Service Discovery that requires the following tags to be present in Consul for the services:
`cluster-port:${AKKA_MANAGEMENT_PORT}` 
`cluster-system-name:some-system`
(of course this could be configurable).

What do you think? Would it be a good thing to add or maybe my approach is wrong?

**Comment by ktoso** [comment]
Disclaimer that we/I are not deeply familiar with consul (which is why such plugin was not provided out of the box really).

When you say tag you mean simply a key-value pair that is registered for a given service name?
That sounds exactly what one would need indeed -- for the IP and port where akka-management's endpoint is to be located.

About the system name note that it should not default to a hardcoded string, but use the current system's name for example.

All in all this sounds good and will be easier to discuss in detail on a PR with code.
I'd be very interested in bringing in support for it, so really looking forward to your PR.
